tests/Zend/Ldap/BindTest.php

```diff
@@ -181,7 +181,7 @@ public function testRequiresDnWithoutDnBind()
         } catch (Zend_Ldap_Exception $zle) {
             /* Note that if your server actually allows anonymous binds this test will fail.
              */
-            $this->assertContains('Failed to retrieve DN', $zle->getMessage());
+            $this->assertContains('No object found for', $zle->getMessage());
```

Comment:
these should probably be version checked:

```php
if (PHP_VERSION_ID >= 80100) ...
```

or do the string differences come from OpenLDAP library?

Reply:
could then check for ldap library version:
- https://github.com/composer/composer/blob/bd6a5019b3bf5edf13640522796f54accaad789e/src/Composer/Repository/PlatformRepository.php
- https://github.com/composer/composer/blob/bd6a5019b3bf5edf13640522796f54accaad789e/src/Composer/Platform/Runtime.php#L91-L102

```
ldap

LDAP Support => enabled
Total Links => 0/unlimited
API Version => 3001
Vendor Name => OpenLDAP
Vendor Version => 20603
SASL Support => Enabled
```

Reply:
my proposal is to just use whatever version you extract as "vendor version" and use it as ">=" and leave the previous strings to the "else" branch.

but the final say is to maintainers, do they want to keep the "older strings" present as well.

simplest and cleanest is just go on and use new strings. 

you could guess by file timestamps (zend tests vs openldap releases) what version it might had been, probably some 10+ year old 2.4.x:
- https://www.openldap.org/software/download/OpenLDAP/openldap-release/

Reply:
I think that LDAP library version (Vendor Version => 20603) or in my case: 20447, does not correspond to OpenLDAP server version:  2.5.13 in my case. And the error messages depend on OpenLDAP server version. I tested it on 2.5 and 2.6 is the latest. It's likely that the old messages were present in version 2.4 of OpenLDAP server like you suggest.

Reply:
I'm sure most of the errors come from client library or php extension.

since there aren't much strings in ext/ldap/ldap.c:
- https://github.com/php/php-src/blob/php-8.2.0/ext/ldap/ldap.c

I am guessing libldap.

for example this particular one can't come from ldap server:
```diff
            $this->assertContains('Can\'t contact LDAP server', $zle->getMessage());
            $this->assertContains('Failed to connect to LDAP server:', $zle->getMessage());
```

maybe only few come from server, like `No object found for`.

Reply:
@glensc 
The error message comes from:
https://github.com/openldap/openldap/blob/5e2fa8a213b8a5d14a1d6aaf6852ad4188db2c3f/libraries/libldap/error.c
I could not find any version with "Failed to connect to LDAP server" in it on github, but new string passes on all php versions it seems.

Reply:
seems maintainers did not voice their opinion here. so can stay as you committed.